src/client/app/shared/canvas/canvas.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable } from '@angular/core';
+import '../../shared/card/card';
+
+// TODO(eyuelt): where should this be defined?
+export interface Point {
+  x:number;
+  y:number;
+}
+
+// NOTE: These type aliases are not type-checked. They are just for readability.
+// TODO(eyuelt): is there a way of getting these type-checked?
+// A point in the coordinate system of the viewport.
+export type ViewportCoord = Point;
+// A point in the coordinate system of the canvas.
+export type CanvasCoord = Point;
+
+/**
+ * The CanvasService maintains the data of the cards to show on the canvas,
+ * handles zooming and panning on the canvas.
+ * Zooming grows and shrinks the viewport relative to the canvas and panning
+ * translates the viewport relative to the canvas.
+ * This class also handles converting between the viewport's coordinate system,
+ * the size of which is specified by viewportSize, and the canvas' coordinate
+ * system, which is infinitely large.
+ * The viewport coord system's origin is at the top left of the viewport.
+ * The canvas coord system's origin starts off at the center of the viewport.
+ */
+@Injectable()
+export class CanvasService {
+  kMinZoomScale: number = 0.1;
+  kMaxZoomScale: number = 10.0;
+  // The models of the cards to show on the canvas.
+  cards: Card[] = [];
```

Comment:
This state should eventually be demoted to some central state management:

https://github.com/ngrx/store